quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/hostname/DefaultHostnameProvider.java

```diff
@@ -237,121 +35,16 @@ public String getId() {
 
     @Override
     public HostnameProvider create(KeycloakSession session) {
-        return this;
+        return provider;
     }
 
     @Override
-    public void init(Config.Scope config) {
-        boolean isHttpEnabled = Boolean.parseBoolean(getConfigValue("kc.http-enabled").getValue());
-        String configPath = getConfigValue("kc.http-relative-path").getValue();
-
-        if (!configPath.startsWith("/")) {
-            configPath = "/" + configPath;
-        }
-
-        String httpsPort = getConfigValue("kc.https-port").getValue();
-        String configPort = isHttpEnabled ? getConfigValue("kc.http-port").getValue() : httpsPort ;
-
-        String scheme = isHttpEnabled ? "http://" : "https://";
-
-        localAdminUri = URI.create(scheme + "localhost:" + configPort + configPath);
-
-        frontEndHostName = config.get("hostname");
-
-        try {
-            String url = config.get("hostname-url");
-
-            if (url != null) {
-                frontEndBaseUri = new URL(url).toURI();
-            }
-        } catch (MalformedURLException | URISyntaxException cause) {
-            throw new RuntimeException("Invalid base URL for FrontEnd URLs: " + config.get("hostname-url"), cause);
-        }
-
-        if (frontEndHostName != null && frontEndBaseUri != null) {
-            throw new RuntimeException("You can not set both '" + HostnameOptions.HOSTNAME.getKey() + "' and '" + HostnameOptions.HOSTNAME_URL.getKey() + "' options");
-        }
-
-        if (config.getBoolean("strict", false) && (frontEndHostName == null && frontEndBaseUri == null)) {
-            throw new RuntimeException("Strict hostname resolution configured but no hostname setting provided");
-        }
-
-        hostnameEnabled = (frontEndHostName != null || frontEndBaseUri != null);
-
-        if (frontEndBaseUri == null) {
-            strictHttps = hostnameEnabled && config.getBoolean("strict-https", false);
+    public void init(Scope config) {
+        if (Profile.isFeatureEnabled(Feature.HOSTNAME_V1)) {
+            this.provider = new V1HostnameProvider().init(config);
```

Comment:
IMHO, this approach feels a bit odd, TBH. Each implementation of an SPI should have its own factory. Maybe we could rather use `EnvironmentDependentProviderFactory`? I think we don't need the explicit exception for when no hostname provider is enable. That should not normally happen as we don't allow disabling this feature.